packages/babel-helpers/src/helpers/applyDecs2311.ts

```diff
@@ -334,100 +334,98 @@ export default /* @no-mangle */ function applyDecs2311(
         }.bind(null, decoratorFinishedRef),
```

Comment:
We don't need to `.bind` here, we can just capture it from the outer scope right? Since `decoratorFinishedRef`. And then we could even just replace `decoratorFinishedRef` with a plain boolean.

Reply:
This produces another observable change.
The `addInitializer` saved by the user in `@dec1` can be used in `@dec2`, which is the current behavior of ts.
I'm fine with both behaviors.

Reply:
I opened https://github.com/microsoft/TypeScript/issues/57300 for TS.